src/Sentry.Unity/NativeUtils/SentryPlatformServices.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+
+namespace Sentry.Unity.NativeUtils;
+
+/// <summary>
+/// These are SDK's services that are only available at runtime and cannot be baked into the SDK. The
+/// <c>SentryInitialization.cs</c> is provided as <c>.cs</c> and gets compiled with the game. It sets <c>IUnityInfo</c>
+/// and the <c>PlatformConfiguration</c> callback during the game's startup so that they are available during initializtion.
+/// </summary>
+/// <remarks>Consider this <c>internal</c>.</remarks>
+public static class SentryPlatformServices
+{
+    private static ISentryUnityInfo? _unityInfo;
+
+    /// <summary>
+    /// The UnityInfo holds methods that rely on conditionally compilation, i.e. IL2CPP backend.
+    /// </summary>
+    public static ISentryUnityInfo UnityInfo
+    {
+        get => _unityInfo ?? throw new InvalidOperationException("UnityInfo is null.");
+        set
+        {
+            if (_unityInfo != null)
+            {
+                throw new InvalidOperationException("Should not set twice. lol.");
```

Comment:
why `lol`? 

We don't want it to be set twice why?
If we want to let the user set but they have to disable auto init. We should say that on the exception message to help the user understand why something went wrong.

I understand it shouldn't happen in any "default" mode of the SDK, but given the many migrations paths, and other init ways we offered in the past, no way users can accidently try to set it twice?

```suggestion
                throw new InvalidOperationException("Should not set twice.");
```

Reply:
You caught my wip push from before the weekend 😅.
I've updated the documentation and exception messages.